sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -794,3 +795,137 @@ def test_staged_predict(HistGradientBoosting, X, y):
 
             assert_allclose(staged_predictions, pred_aux)
             assert staged_predictions.shape == pred_aux.shape
+
+
+@pytest.mark.parametrize("insert_missing", [False, True])
+@pytest.mark.parametrize("make_dataset, Est, metric", [
+    (make_regression, HistGradientBoostingRegressor, mean_squared_error),
+    (make_classification, HistGradientBoostingClassifier, r2_score)
```

Comment:
what was the reasoning for checking r2_score on a classification task?

Note that I tried using the `.score()` but the test doesn't pass.

Checking that est_cat does better than est_no_cat is a great test, but I'm not sure we should expect better results with the current dataset, since it comes from `make_classification / make_regression`, where the target depends on ordinal features. Introducing categories here means that we make the features non-ordinal, but they're still ordinal as far as the target is concerned. So I would actually expect est_no_cat to do better than est_cat.

It would be better IMHO to have a more thorough check where categories aren't introduced "randomly"

Reply:
> what was the reasoning for checking r2_score on a classification task?

There was no particular reason.

I was using this test to make sure things did not go horribly wrong when working on the splitter + predictor. Maybe we can just place a lower bound on the metric?

Reply:
I pushed a test in  51cb926  which I believe is a more sensible functional test for comparing prediction performance. I edited this one in  4cacf4368b4c1390c01676a81280abbc1760dc08 so that it's only about checking nans and unknown categories